cluster-autoscaler/core/scale_up.go

```diff
@@ -475,36 +498,54 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 		return &status.ScaleUpStatus{
 				ScaledUp:                true,
 				ScaleUpInfos:            scaleUpInfos,
-				PodsRemainUnschedulable: getRemainingPods(podsRemainUnschedulable),
+				PodsRemainUnschedulable: getRemainingPods(podsRemainUnschedulable, skippedNodeGroups),
 				PodsTriggeredScaleUp:    bestOption.Pods,
 				PodsAwaitEvaluation:     getPodsAwaitingEvaluation(unschedulablePods, podsRemainUnschedulable, bestOption.Pods)},
 			nil
 	}
 
-	return &status.ScaleUpStatus{ScaledUp: false, PodsRemainUnschedulable: getRemainingPods(podsRemainUnschedulable)}, nil
+	return &status.ScaleUpStatus{ScaledUp: false, PodsRemainUnschedulable: getRemainingPods(podsRemainUnschedulable, skippedNodeGroups)}, nil
 }
 
-func getRemainingPods(podSet map[*apiv1.Pod]bool) []*apiv1.Pod {
-	result := make([]*apiv1.Pod, 0)
-	for pod, val := range podSet {
-		if val {
-			result = append(result, pod)
+func getRemainingPods(schedulingErrors map[*apiv1.Pod][]*simulator.PredicateError, skipped int) map[*apiv1.Pod]string {
+	remaining := map[*apiv1.Pod]string{}
+	for pod, errs := range schedulingErrors {
+		messages := []string{}
+		aggregated := map[string]int{}
+		for _, err := range errs {
+			// Aggregate by reason, since predicate name isn't as helpful.
+			// If two different predicates fail with the same reason, so be it.
+			for _, reason := range err.Reasons() {
+				aggregated[reason]++
+			}
+		}
+		for msg, count := range aggregated {
+			messages = append(messages, fmt.Sprintf("%s (%d)", msg, count))
 		}
+		if skipped > 0 {
+			messages = append(messages, fmt.Sprintf("can't increase node group size (%d)", skipped))
+		}
+		remaining[pod] = strings.Join(messages, ", ")
```

Comment:
As far as I can see we only do the whole string formatting so we can use it in EventingScaleUpStatusProcessor. Maybe we should move this to a utility function and call it from inside the processor? In general a data structure may be more useful for some custom processor than a string.

Reply:
Agreed, this is one of the points for potential improvement mentioned in the description. 

I can work on it now, but it'll make this PR even bigger (or hacky - e.g. I could add an artificial PredicateError for skipped groups, instead of yet-another-custom-struct).